src/map.js

```diff
@@ -1716,12 +1739,24 @@ var map = function (arg) {
   }
 
   /**
-   * Return the nearest valid bounds maintaining the
-   * width and height. Does nothing if m_clampBounds* is
-   * false.
+   * Return the nearest valid bounds maintaining the width and height.  Does
```

Comment:
I am not entire sure about these options and will need some discussion to understand it completely.